### DISCUSSION TITLE
编译 错误 2

### DISCUSSION THREAD

**Discussion by aepub** [immortalwrt/immortalwrt#1459] [Q&A]
### Describe the bug

已经编译成功过，刷进小主机正常使用，但是为了增加ddns-go和lede的wiregaude模块，重新进行编译，但出现以下错误：
install -d -m0755 /home/imm/immortalwrt2305/tmp/stage-lua/usr/lib
cp -fpR /home/imm/immortalwrt2305/build_dir/target-x86_64_musl/lua-5.1.5/ipkg-install/usr/lib/liblua.{a,so*} /home/imm/immortalwrt2305/tmp/stage-lua/usr/lib/
ln -sf liblua.so.5.1.5 /home/imm/immortalwrt2305/tmp/stage-lua/usr/lib/liblualib.so
install -d -m0755 /home/imm/immortalwrt2305/tmp/stage-lua/usr/lib/pkgconfig
cp -fpR /home/imm/immortalwrt2305/build_dir/target-x86_64_musl/lua-5.1.5/etc/lua.pc /home/imm/immortalwrt2305/tmp/stage-lua/usr/lib/pkgconfig/
find /home/imm/immortalwrt2305/tmp/stage-lua -name '*.la' | xargs -r rm -f;
if [ -f /home/imm/immortalwrt2305/staging_dir/target-x86_64_musl/packages/lua.list ]; then /home/imm/immortalwrt2305/scripts/clean-package.sh "/home/imm/immortalwrt2305/staging_dir/target-x86_64_musl/packages/lua.list" "/home/imm/immortalwrt2305/staging_dir/target-x86_64_musl"; fi
if [ -d /home/imm/immortalwrt2305/tmp/stage-lua ]; then (cd /home/imm/immortalwrt2305/tmp/stage-lua; find ./ > /home/imm/immortalwrt2305/tmp/stage-lua.files);  SHELL= flock /home/imm/immortalwrt2305/tmp/.staging-dir.flock -c ' mv /home/imm/immortalwrt2305/tmp/stage-lua.files /home/imm/immortalwrt2305/staging_dir/target-x86_64_musl/packages/lua.list && cp -fpR /home/imm/immortalwrt2305/tmp/stage-lua/* /home/imm/immortalwrt2305/staging_dir/target-x86_64_musl/; '; fi
rm -rf /home/imm/immortalwrt2305/tmp/stage-lua
touch /home/imm/immortalwrt2305/staging_dir/target-x86_64_musl/stamp/.lua_installed
rm -rf /home/imm/immortalwrt2305/build_dir/target-x86_64_musl/lua-5.1.5/.pkgdir/lua.installed /home/imm/immortalwrt2305/build_dir/target-x86_64_musl/lua-5.1.5/.pkgdir/lua
mkdir -p /home/imm/immortalwrt2305/build_dir/target-x86_64_musl/lua-5.1.5/.pkgdir/lua
install -d -m0755 /home/imm/immortalwrt2305/build_dir/target-x86_64_musl/lua-5.1.5/.pkgdir/lua/usr/bin
install -m0755 /home/imm/immortalwrt2305/build_dir/target-x86_64_musl/lua-5.1.5/ipkg-install/usr/bin/lua5.1 /home/imm/immortalwrt2305/build_dir/target-x86_64_musl/lua-5.1.5/.pkgdir/lua/usr/bin/
ln -sf lua5.1 /home/imm/immortalwrt2305/build_dir/target-x86_64_musl/lua-5.1.5/.pkgdir/lua/usr/bin/lua
touch /home/imm/immortalwrt2305/build_dir/target-x86_64_musl/lua-5.1.5/.pkgdir/lua.installed
mkdir -p /home/imm/immortalwrt2305/staging_dir/target-x86_64_musl/root-x86/stamp
SHELL= flock /home/imm/immortalwrt2305/tmp/.root-copy.flock -c 'cp -fpR /home/imm/immortalwrt2305/build_dir/target-x86_64_musl/lua-5.1.5/.pkgdir/lua/. /home/imm/immortalwrt2305/staging_dir/target-x86_64_musl/root-x86/'
touch /home/imm/immortalwrt2305/staging_dir/target-x86_64_musl/root-x86/stamp/.lua_installed
echo "lua" >> /home/imm/immortalwrt2305/staging_dir/target-x86_64_musl/pkginfo/lua.default.install
make[3]: Leaving directory '/home/imm/immortalwrt2305/package/utils/lua'
time: package/utils/lua/compile#4.82#0.61#5.40
make[2]: Leaving directory '/home/imm/immortalwrt2305'
make[1]: *** [package/Makefile:123: /home/imm/immortalwrt2305/staging_dir/target-x86_64_musl/stamp/.package_compile] Error 2
make[1]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/imm/immortalwrt2305'
make[1]: Leaving directory '/home/imm/immortalwrt2305'
make: *** [/home/imm/immortalwrt2305/include/toplevel.mk:232：world] 错误 2

### ImmortalWrt version

openwrt-23.05

### ImmortalWrt release

openwrt-23.05

### ImmortalWrt target/subtarget

openwrt-23.05

### Device

X86/64

### Image kind

Self-built image

### Terms

- [X] I am reporting an issue for ImmortalWrt, not an unsupported fork.

**Comment by 1715173329** [comment]
1. 单线程编译
2. 自行添加插件遇到问题不在本项目处理范围内

**Comment by aepub** [comment]
主要是用官方的23或者24版软件包安装wireadgue时提示内核版本不对，所有也只能自己编译，-j1编译后还是提示错误：
gcc (Ubuntu 9.5.0-1ubuntu1~22.04) 9.5.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

./tool/file2lastrev.rb:6:in `require': cannot load such file -- optparse (LoadError)
        from ./tool/file2lastrev.rb:6:in `<main>'
make[4]: *** [uncommon.mk:1225: .revision.time] Error 1
make[4]: Leaving directory '/home/imm/immortalwrt2305/build_dir/hostpkg/ruby-3.2.5'
make[3]: *** [Makefile:1162: /home/imm/immortalwrt2305/build_dir/hostpkg/ruby-3.2.5/.built] Error 2
make[3]: Leaving directory '/home/imm/immortalwrt2305/feeds/packages/lang/ruby'
time: package/feeds/packages/ruby/host-compile#0.30#0.11#0.45
    ERROR: package/feeds/packages/ruby [host] failed to build.
make[2]: *** [package/Makefile:127: package/feeds/packages/ruby/host/compile] Error 1
make[2]: Leaving directory '/home/imm/immortalwrt2305'
make[1]: *** [package/Makefile:123: /home/imm/immortalwrt2305/staging_dir/target-x86_64_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/home/imm/immortalwrt2305'
make: *** [/home/imm/immortalwrt2305/include/toplevel.mk:232：world] 错误 2

**Reply by 1715173329** [reply]
openwrt/packages#24450 immortalwrt/packages#1251
